File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        int slow = nums[0], fast = nums[nums[0]];
        while (fast != slow){
            fast = nums[nums[fast]];
            slow = nums[slow];
        }
        fast = 0;
        while (fast != slow){
            slow = nums[slow];
            fast = nums[fast];
        }
        return fast;
    }
}

File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int l = 0, r = n - 1; 
        int i = 0;
        while (i <= r) {
            if (nums[i] == 0) {
                // put it in left;
                swap(nums[l++], nums[i]);
                i++;
            } else if (nums[i] == 2) {
                // put it in right;
                swap(nums[i], nums[r--]);
            } else {
                i++;
            }
        }
    }
};